File: 0643-maximum-average-subarray-i/0643-maximum-average-subarray-i.js
/**
 * @param {number[]} nums
 * @param {number} k
 * @return {number}
 */
var findMaxAverage = function(nums, k) {
    let average=-Number.MAX_VALUE
    let current=0
    let sum=0
    for(let i=0;i<nums.length;i++){
        sum+=nums[i]
        if(i>=k-1){
            current=sum/k
            average=Math.max(average,current)
            sum-=nums[i-(k-1)]
        }
    }
    return average
};

File: 0643-maximum-average-subarray-i/0643-maximum-average-subarray-i.py
class Solution:
    def findMaxAverage(self, nums: List[int], k: int) -> float:
        average=-inf
        current=0
        sum=0
        for i in range(len(nums)):
            sum+=nums[i]
            if i>=k-1:
                current=sum/k
                average=max(average,current)
                sum-=nums[i-(k-1)]
        return average